src/activation.jl

```diff
@@ -17,6 +17,15 @@ const sigmoid = σ
   σ(x::ForwardDiff.Dual{T,Float32}) where T = σ_stable(x)
 end
 
+"""
+   hardσ(x, a=0.2) = max(0, min(1.0, a * x + 0.5))
+
+Segment-wise linear approximation of sigmoid
+See: [BinaryConnect: Training Deep Neural Networks withbinary weights during propagations](https://arxiv.org/pdf/1511.00363.pdf)
+"""
+hardσ(x::Real, a=0.2) = oftype(x/1, max(zero(x/1), min(one(x/1), oftype(x/1,a) * x + oftype(x/1,0.5))))
```

Comment:
I think division by `1` should be replaced by division by `one(x)` like fo other activations

Reply:
Actually one(x) or 1 doesn't make much difference here. In both the cases output precision remains the same. Moreover, division by is faster than division by one(x)
```
julia> x = Float32(1)
1.0f0

julia> @time x/one(x)
  0.002585 seconds (28 allocations: 1.875 KiB)
1.0f0

julia> @time x/1
  0.000009 seconds (5 allocations: 176 bytes)
1.0f0
```